scrapper.py

```diff
@@ -26,38 +35,122 @@ class Crawler:
     Crawler implementation
     """
     def __init__(self, seed_urls, max_articles: int):
-        pass
+        self.seed_urls = seed_urls
+        self.max_articles = max_articles
+        self.urls = []
 
     def _extract_url(self, article_bs):
-        pass
-
+        main_bs = article_bs.find('div', {'class': 'article-tile-news'})
+        for url_bs in main_bs:
+            if len(self.urls) < self.max_articles:
+                link = url_bs.find('a')
+                full_link = link['href']
+                self.urls.append(HTTP_MAIN + full_link)
+
+    # if 'https://esquire.ru/' + url_bs['href'] not in self.urls:
+    # urls_bs = main_bs.find_all('а')
     def find_articles(self):
         """
         Finds articles
         """
-        pass
+        for seed_url in self.seed_urls:
+            response = requests.get(url=seed_url, headers=HEADERS)
+            if not response.ok:
+                continue
+            soup = BeautifulSoup(response.text, 'lxml')
+            self._extract_url(soup)
 
     def get_search_urls(self):
         """
         Returns seed_urls param
         """
         pass
 
+class HTMLParser:
+    def __init__(self, article_url, article_id):
+        self.article_url = article_url
+        self.article_id = article_id
+        self.article = Article(self.article_url, self.article_id)
+
+    def _fill_article_with_text(self, article_bs):
+        text = article_bs.find('div', {'class': 'article-detail__content'}).text
+        self.article.text = text
+
+    def _fill_article_with_meta_information(self, article_bs):
+        date = article_bs.find('div', {'class': 'breadcrumbs__date'})
+        date_for_meta = datetime.strptime(date.text, '%Y-%m-%d %H:%M')
+        self.article.date = date_for_meta
+
+        author = article_bs.find('div', {'class': 'author-wrapper article-detail__author'})
+        if not author:
+            author = 'Not Found'
+        self.article.author = author
+
+        title = article_bs.find('h1', {'class': 'article-detail__title'}).text
+        self.article.title = title
+
+    def parse(self):
+        response = requests.get(self.article_url, HEADERS)
+        article_bs = BeautifulSoup(response.text, 'lxml')
+
+        self._fill_article_with_text(article_bs)
+        self._fill_article_with_meta_information(article_bs)
+        return self.article
+
 
 def prepare_environment(base_path):
     """
     Creates ASSETS_PATH folder if not created and removes existing folder
     """
-    pass
+    path = Path(base_path)
+    if path.exists():
+        shutil.rmtree(base_path)
+    path.mkdir(parents=True, exist_ok=True)
 
 
 def validate_config(crawler_path):
     """
     Validates given config
     """
-    pass
+    with open(crawler_path) as file:
+        config = json.load(file)
+    seed_urls = config['seed_urls']
+    total_articles = config["total_articles_to_find_and_parse"]
+
+    for seed_url in seed_urls:
```

Comment:
What if someone tries to pass a non-iterable value as seed_urls? For example, an integer.

Reply:
I see that you do check it below. However, if such situation happens, TypeError will be thrown before IncorrectURLError.
Order of checks matters :)